tests/test_optimizer_factory.py

```diff
@@ -72,7 +72,7 @@ def _load_coca() -> FSDP:
 
 # number of parameters for each optimizer group
 GPT2_LINEAR = 66130944
-GPT2_EMBEDDING = 768 * (50304 + 2048)  # n_embd * (vocab_size + block_size)
+GPT2_EMBEDDING = 768 * (50304 + 2048 - 1)  # n_embd * (vocab_size + block_size - 1)
```

Comment:
why `-1`?

Reply:
Because `num_embeddings = block size - 1`, see https://github.com/Modalities/modalities/blob/ab354bb1d7a1e28ccc0ab311d97c246bec4b59af/src/modalities/models/gpt2/gpt2_model.py#L447 This change was introduced in 19d85e3.

Reply:
ah yeah that makes sense. Thanks for the clarification!

Reply:
This solution might be suboptimal though. I opened an issue, see https://github.com/Modalities/modalities/issues/156